### PR TITLE
Add mingw-w64-groonga package

### PR DESCRIPTION
**Pull request by cosmo0920**
Groonga is an opensource fulltext search engine.

### CODE REVIEW

File: mingw-w64-groonga/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    --sbindir=${MINGW_PREFIX}/bin \
+    --with-default-encoding=utf8 \
+    --disable-libedit \
+    --with-zlib \
+    --with-lz4 \
+    --without-stemmer \
+    --with-zeromq \
+    --with-onigmo=system \
+    --enable-shared=yes \
+    --enable-static=yes \
```
Comment (kou) [inline mingw-w64-groonga/PKGBUILD]: We can omit this because this is the default.

File: mingw-w64-groonga/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    --disable-benchmark \
+    --disable-groonga-httpd \
+    --with-mruby
+
+    make
+}
+
+package() {
+    cd $srcdir/${_realname}-$pkgver
+    make DESTDIR="$pkgdir" install
```
Comment (kou) [inline mingw-w64-groonga/PKGBUILD]: Should we adjust path in `.pc` here like this?

https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-arrow/PKGBUILD#L95-L100

File: mingw-w64-groonga/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=9.0.1
+pkgrel=1
+pkgdesc="An opensource fulltext search engine."
+arch=('any')
+url="http://groonga.org/"
+license=('LGPL2')
+source=("http://packages.groonga.org/source/${_realname}/${_realname}-$pkgver.tar.gz")
+depends=("${MINGW_PACKAGE_PREFIX}-zlib"
+         "${MINGW_PACKAGE_PREFIX}-glib2"
```
Comment (kou) [inline mingw-w64-groonga/PKGBUILD]: Do we need this?

File: mingw-w64-groonga/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    --with-mruby
+
+    make
+}
+
+package() {
+    cd $srcdir/${_realname}-$pkgver
+    make DESTDIR="$pkgdir" install
+}
+sha256sums=('f1b14f0fd62ca7da550c11e164edcbcac7d83de75474cc4d3a6574adf2733507')
```
Comment (kou) [inline mingw-w64-groonga/PKGBUILD]: Should we put this here?
Normally, we put `sha256sums()` before function definitions.

File: mingw-w64-groonga/PKGBUILD (hunk 2 above)
Comment (cosmo0920) [reply mingw-w64-groonga/PKGBUILD]: Fixed at https://github.com/msys2/MINGW-packages/pull/5151/commits/3546075a6e1108eaabfbee199f7c24a1b465d12e.

File: mingw-w64-groonga/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    --target=${MINGW_CHOST} \
+    --localstatedir=${MINGW_PREFIX}/var \
+    --sysconfdir=${MINGW_PREFIX}/etc \
+    --sbindir=${MINGW_PREFIX}/bin \
+    --with-default-encoding=utf8 \
+    --disable-libedit \
+    --with-zlib \
+    --with-lz4 \
+    --without-stemmer \
+    --with-zeromq \
```
[the remark this answers is not in the document]
Comment (cosmo0920) [reply mingw-w64-groonga/PKGBUILD]: Added libevent in https://github.com/msys2/MINGW-packages/pull/5151/commits/ad993c1d3997c5ebea1c6e624ffed7b0a1b0b132.

File: mingw-w64-groonga/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+         "${MINGW_PACKAGE_PREFIX}-msgpack-c"
+         "${MINGW_PACKAGE_PREFIX}-onigmo"
+         "${MINGW_PACKAGE_PREFIX}-arrow"
+         "${MINGW_PACKAGE_PREFIX}-openssl")
+makedepends=("${MINGW_PACKAGE_PREFIX}-gcc")
+
+build() {
+    cd $srcdir/${_realname}-$pkgver
+    ./configure \
+    --prefix=${MINGW_PREFIX} \
```
[the remark this answers is not in the document]
Comment (cosmo0920) [reply mingw-w64-groonga/PKGBUILD]: Depends on zstd at https://github.com/msys2/MINGW-packages/pull/5151/commits/b19539d7295cbd3efa452a4d2746adf6cc313f47.

File: mingw-w64-groonga/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+source=("http://packages.groonga.org/source/${_realname}/${_realname}-$pkgver.tar.gz")
+depends=("${MINGW_PACKAGE_PREFIX}-zlib"
+         "${MINGW_PACKAGE_PREFIX}-glib2"
+         "${MINGW_PACKAGE_PREFIX}-zeromq"
+         "${MINGW_PACKAGE_PREFIX}-lz4"
+         "${MINGW_PACKAGE_PREFIX}-pcre"
+         "${MINGW_PACKAGE_PREFIX}-msgpack-c"
+         "${MINGW_PACKAGE_PREFIX}-onigmo"
+         "${MINGW_PACKAGE_PREFIX}-arrow"
+         "${MINGW_PACKAGE_PREFIX}-openssl")
```
[the remark this answers is not in the document]
Comment (cosmo0920) [reply mingw-w64-groonga/PKGBUILD]: Depends on zstd at https://github.com/msys2/MINGW-packages/pull/5151/commits/b19539d7295cbd3efa452a4d2746adf6cc313f47.

[the remark this answers is not in the document]
Comment (cosmo0920) [reply mingw-w64-groonga/PKGBUILD]: Depends on zstd at https://github.com/msys2/MINGW-packages/pull/5151/commits/b19539d7295cbd3efa452a4d2746adf6cc313f47.

File: mingw-w64-groonga/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+pkgdesc="An opensource fulltext search engine."
+arch=('any')
+url="http://groonga.org/"
+license=('LGPL2')
+options=('strip' 'staticlibs')
+source=("http://packages.groonga.org/source/${_realname}/${_realname}-$pkgver.tar.gz")
+depends=("${MINGW_PACKAGE_PREFIX}-zlib"
+         "${MINGW_PACKAGE_PREFIX}-lz4"
+         "${MINGW_PACKAGE_PREFIX}-zstd"
+         "${MINGW_PACKAGE_PREFIX}-pcre"
```
Comment (kou) [inline mingw-w64-groonga/PKGBUILD]: We don't need PCRE when we don't build groonga-httpd.

File: mingw-w64-groonga/PKGBUILD (hunk 7 above)
[the remark this answers is not in the document]
Comment (kou) [reply mingw-w64-groonga/PKGBUILD]: We don't need OpenSSL when we don't build groonga-httpd.

**Comment by cosmo0920** [comment]
Thanks!!

**Pull request merged**